src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala

```diff
@@ -572,13 +568,20 @@ object EthJsonMethodsImplicits extends JsonMethodsImplicits {
       data = data.getOrElse(ByteString("")))
   }
 
+  private def optionalQuantity(input: JValue): Either[JsonRpcError, Option[BigInt]] =
+    input match {
+      case JNothing => Right(None)
+      case o => extractQuantity(o).map(Some(_))
+    }
+
   implicit val eth_getAccountRecentTransactions = new JsonDecoder[GetAccountRecentTransactionsRequest] with JsonEncoder[GetAccountRecentTransactionsResponse] {
     def decodeJson(params: Option[JArray]): Either[JsonRpcError, GetAccountRecentTransactionsRequest] =
       params match {
-        case Some(JArray((address: JString) :: Nil)) =>
+        case Some(jarray: JArray) =>
```

Comment:
Very minor detail: Maybe this could be changed to what's done in [`eth_sign`](https://github.com/input-output-hk/mantis/blob/9b55ae760559f60dc515a235af4b2b1a376ce72c/src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala#L541) for extracting the parameters?

Reply:
maybe I'm missing something, but is this possible for variable-sized Jarray (either 1 or 2 elems)?

Reply:
You are right, it doesn't seem to allow doing so.
The only option seems to be something like https://github.com/input-output-hk/mantis/blob/phase/daedalus/src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala#L255, but I don't have any strong feelings regarding if doing so is necessary or not, we could leave it as is.